app/services/twitter/tweet_parser_service.rb

```diff
@@ -3,7 +3,8 @@ class Twitter::TweetParserService < Twitter::WebhooksBaseService
 
   def perform
     set_inbox
-    return if message_already_exist? || user_has_blocked?
+
+    return if message_already_exist? || user_has_blocked? || !tweets_enabled?
```

Comment:
!tweets_enabled? could be first condition so that we won't have to make another DB call to check whether the message exists